src/htcdaskgateway/cluster.py

```diff
@@ -25,15 +25,27 @@ def __init__(self, **kwargs):
         self.batchWorkerJobs = []
         self.defaultImage = 'coffeateam/coffea-base-almalinux8:0.7.22-py3.10'
         self.cluster_options = kwargs.get('cluster_options')
+        if 'image_registry' in kwargs:
+            self.image_prefix = kwargs.pop('image_registry')
+        else:
+            self.image_prefix = 'registry.hub.docker.com'
         
         #set default image if the image is not specified by user
         if not kwargs.get('image') and (not self.cluster_options or not self.cluster_options.image):
             kwargs['image'] = self.defaultImage
             print("Selected Image: ", kwargs['image'])
             self.condor_image = self.defaultImage
         else:
+            print("Selected Image: ", kwargs['image'])
             self.condor_image = kwargs.get('image')
-        
+            
+        if self.image_prefix is not 'registry.hub.docker.com':
```

Comment:
Why this condition - shouldn't we execute the next line always?

Reply:
I don't think it has to be there, it was more just because I had seen an issue at some point where adding the prefix for the default seemed to make it hang. But I can try without it, it also doesn't necessarily need to run for public docker images.

Reply:
Oh, that's right.  When we do `docker pull` it needs to *not* have the registry.hub.docker.com.

Reply:
I am actually not entirely sure about that, I have run it with the registry.hub.docker.com on accident and it ran ok, but I have seen it hang a few times before. I can try to test without the if statement and see if it is consistently deploying. It's possible a different issue was causing the hang up